aiida_codtools/workflows/cif_clean.py

```diff
@@ -0,0 +1,163 @@
+# -*- coding: utf-8 -*-
+from aiida.common.extendeddicts import AttributeDict
+from aiida.orm import Code, Group
+from aiida.orm.data.cif import CifData
+from aiida.orm.data.base import Float, Str
+from aiida.orm.data.parameter import ParameterData
+from aiida.orm.data.structure import StructureData
+from aiida.orm.utils import CalculationFactory
+from aiida.tools import get_kpoints_path
+from aiida.work.workchain import WorkChain, ToContext
+from aiida.work.workfunctions import workfunction
+
+
+CifFilterCalculation = CalculationFactory('codtools.cif_filter')
+CifSelectCalculation = CalculationFactory('codtools.cif_select')
+
+
+class CifCleanWorkChain(WorkChain):
+    """
+    WorkChain to clean a CifData node using the cif_filter and cif_select scripts of cod-tools
+    It will first run cif_filter to correct syntax errors, followed by cif_select which will remove
+    various tags and canonicalize the remaining tags
+    """
+
+    ERROR_CIF_FILTER_FAILED = 1
+    ERROR_CIF_SELECT_FAILED = 2
+
+    @classmethod
+    def define(cls, spec):
+        super(CifCleanWorkChain, cls).define(spec)
+        spec.input('cif', valid_type=CifData,
+            help='the CifData node that is to be cleaned')
+        spec.input('cif_filter', valid_type=Code,
+            help='the AiiDA code object that references the cod-tools cif_filter script')
+        spec.input('cif_select', valid_type=Code,
+            help='the AiiDA code object that references the cod-tools cif_select script')
+        spec.input('options', valid_type=ParameterData,
+            help='options for the calculations')
+        spec.input('tags', valid_type=Str, default=Str('_publ_author_name,_citation_journal_abbrev'),
+            help='comma separated tag names that are to be removed by the cif_select step')
+        spec.input('symprec', valid_type=Float, default=Float(5E-3),
+            help='the symmetry precision used by SeeK-path for crystal symmetry refinement')
+        spec.input('group_cif', valid_type=Group, required=False, non_db=True,
+            help='an optional Group node to which the final cleaned CifData node will be added')
+        spec.input('group_structure', valid_type=Group, required=False, non_db=True,
+            help='an optional Group node to which the final reduced StructureData node will be added')
+        spec.outline(
+            cls.run_filter_calculation,
+            cls.inspect_filter_calculation,
+            cls.run_select_calculation,
+            cls.inspect_select_calculation,
+            cls.create_reduced_structure,
```

Comment:
create_reduced_structure: do it optional? (when e.g. group_structure is not defined)